### ISSUE TITLE
Engine BE 4: Duplicate routes and controllers

### ISSUE THREAD

**Issue by kayline** [department-of-veterans-affairs/va.gov-team#98358]
## Value Statement

**_As a_** backend dev

**_I want to_** duplicate our routes and controllers in the engine
**_So that_** we can start moving away from the non-engine versions and further isolate our code

---

## Background Context

List of URIs and controllers:
- [x] /v1/nod_callbacks v1/decision_review_notification_callbacks#create {:format=>"json"}
- [x] /v1/decision_review_evidence(.:format) v1/decision_review_evidences#create {:format=>"json"}
- [x] /v1/higher_level_reviews/contestable_issues(/:benefit_type)(.:format) v1/higher_level_reviews/contestable_issues#index {:format=>"json"}
- [x] /v1/notice_of_disagreements/contestable_issues(.:format) v1/notice_of_disagreements/contestable_issues#index {:format=>"json"}
- [x] /v1/supplemental_claims/contestable_issues(/:benefit_type)(.:format) v1/supplemental_claims/contestable_issues#index {:format=>"json"}
- [x] /v1/higher_level_reviews(.:format) v1/higher_level_reviews#create {:format=>"json"}
- [x] /v1/higher_level_reviews/:id(.:format) v1/higher_level_reviews#show {:format=>"json"}
- [x] /v2/higher_level_reviews(.:format) v2/higher_level_reviews#create {:format=>"json"}
- [x] /v2/higher_level_reviews/:id(.:format) v2/higher_level_reviews#show {:format=>"json"}
- [x] /v1/notice_of_disagreements(.:format) v1/notice_of_disagreements#create {:format=>"json"}
- [x] /v1/notice_of_disagreements/:id(.:format) v1/notice_of_disagreements#show {:format=>"json"}
- [x] /v1/supplemental_claims(.:format) v1/supplemental_claims#create {:format=>"json"}
- [x] /v1/supplemental_claims/:id(.:format) v1/supplemental_claims#show {:format=>"json"}

## Tasks

- [x] Duplicate controller files
- [x] Duplicate controller or request specs
- [x] Duplicate necessary services/helpers
- [x] Add new routes to engine

## Dependencies 

- Wait until #98352 is complete and working, but not blocked by #98355 

## Acceptance Criteria

- [ ] _AC_
- [ ] _AC_
- [ ] _AC_

## Out of scope

- Add here

## Open questions

- Add here

## Designs and Build Notes

- Add here

## Outcome, Success Measure, KPI(S), and Tracking Link

- Add here

### Enablement team (if needed)
@ member of team

## Definition of Ready

- [ ] Clear value description
- [ ] Testable acceptance criteria
- [ ] Accessibility added to acceptance criteria
- [ ] Approved designs attached
- [ ] Sample data provided where appropriate
- [ ] Estimated to fit within the sprint
- [ ] Dependencies and blockers linked

## Definition of Done

- [ ] Meets acceptance criteria
- [ ] Passed E2E testing (90% coverage)
- [ ] Passed unit testing (90% coverage)
- [ ] Passed integration testing (if applicable)
- [ ] Code reviewed (internal)
- [ ] Submitted to staging
- [ ] Team approved production verification process
- [ ] Reviewed and approved by product and/or design

**Comment by comaurice** [comment]
Waiting for platform review

**Comment by dfong-adh** [comment]
Deployed and verified in staging.

**Comment by comaurice** [comment]
Tested in staging, needs verification in Prod.
